build_runner_core/lib/src/asset_graph/graph.dart

```diff
@@ -374,17 +374,20 @@ class AssetGraph {
   bool _actionMatches(BuildAction action, AssetId input) {
     if (input.package != action.package) return false;
     if (!action.generateFor.matches(input)) return false;
-    Iterable<String> inputExtensions;
+
     if (action is InBuildPhase) {
-      inputExtensions = action.builder.buildExtensions.keys;
+      if (expectedOutputs(action.builder, input).isEmpty) return false;
     } else if (action is PostBuildAction) {
-      inputExtensions = action.builder.inputExtensions;
+      // todo: Do we want to allow capture groups for post process builders?
+      // I guess not?
```

Comment:
They can already output whatever they want, but it would allow some more interesting matching on input extensions....

If it isn't supported (fine for now imo), we should throw if we see them so people know it isn't supported.